x/tally/keeper/abci.go

```diff
@@ -117,6 +113,9 @@ func (k Keeper) ProcessTallies(ctx sdk.Context, coreContract sdk.AccAddress) err
 				dataResults[i].ExitCode = uint32(result.exitInfo.ExitCode)
 				dataResults[i].Result = result.result
 			}
+			dataResults[i].Consensus = result.consensus
+			dataResults[i].GasUsed = result.execGasUsed + result.tallyGasUsed
```

Comment:
We should always set the gas used. If the requestor specified a wrong tally program or incorrect filter they still should pat for the computation that was used for their request.

Reply:
I just thought it makes more sense to check the value of `err` right after its assignment as it's always done in golang.

Reply:
Aaah, on mobile it looked like it was moved into the `else` clause. Never mind.